dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostFactoryImpl.java

```diff
@@ -791,6 +791,17 @@ protected Optional<List<Host>> search(final String siteNameFilter, final String
             hostList.addAll(APILocator.getPermissionAPI().filterCollection(siteList, PermissionAPI
                     .PERMISSION_READ, respectFrontendRoles, user));
             hostList = hostList.stream().limit(limit).collect(Collectors.toList());
+
+            //We want to include the system host in the list of hosts
+            final Host systemHost = APILocator.systemHost();
+            //Check if we need to include it, if we have the permissions and if it is in the list.
+            final boolean sysHostPermission = showSystemHost
+                    && APILocator.getPermissionAPI().doesSystemHostHavePermissions(systemHost, user, respectFrontendRoles, Host.class.getCanonicalName())
+                    && !hostList.contains(systemHost);
```

Comment:
Shouldn't you switch this statement with the one above so that you don't call the `doesSystemHostHavePermissions` method if the System Host is already in the `hostList` ?

Reply:
Done, thanks for the comment!